src/tools/scale_tool.cpp

```diff
@@ -100,8 +112,17 @@ void ScaleTool::dragMove()
 	
 	auto scaling_length = (cur_pos_map - scaling_center).length();
 	scaling_factor = qMax(minimum_length, scaling_length) / qMax(minimum_length, reference_length);
-	for (auto object : editedObjects())
-		object->scale(scaling_center, scaling_factor);
+
+	if (using_selection_center)
+	{
+		for (auto object : editedObjects())
```

Comment:
Here and in the loop below, I would be glad if you could turn this into `auto object*`.

Reply:
I recall checking this exact place and I tend to believe `editedObjects` returns iterator and not a pointer. I'll study the interactions again.

Reply:
Don't get sidetracked by the `ObjectsRange` class: It is the `for` loop which derefences the iterators it operates on internally.

Reply:
Changed.